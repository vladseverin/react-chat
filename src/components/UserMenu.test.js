import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UserMenu from './UserMenu';

const mockProps = {
  disabled: false,
  activeUser: {
    firstName: 'Vladislav',
    lastName: 'Severin',
    username: 'vlad',
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  logout: jest.fn(),
  editUser: jest.fn(),
};

describe('<UserMenu />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserMenu {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<UserMenu {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
