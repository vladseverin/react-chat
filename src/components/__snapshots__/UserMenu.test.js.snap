// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenu /> renders correctly 1`] = `
<button
  aria-haspopup="true"
  aria-owns={null}
  className="MuiButtonBase-root-37 MuiIconButton-root-31 MuiIconButton-colorInherit-32"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-36"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-40"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
        />
      </g>
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-46"
  />
</button>
`;
